src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,184 @@
+import "@polymer/paper-listbox/paper-listbox";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  customElement,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import computeStateName from "../../common/entity/compute_state_name";
+import {
+  DeviceTrigger,
+  fetchDeviceTriggers,
+  triggersEqual,
+} from "../../data/device_automation";
+
+@customElement("ha-device-trigger-picker")
+class HaDeviceTriggerPicker extends LitElement {
+  public hass?: HomeAssistant;
+  @property() public label?: string;
+  @property() public deviceId?: string;
+  @property() public triggers: { [key: string]: DeviceTrigger } = {};
+  public presetTrigger?: DeviceTrigger;
+
+  private noTrigger: DeviceTrigger = {
+    device_id: "",
+    platform: "device",
+    domain: "",
+    entity_id: "",
+  };
+  private unknownTrigger?: DeviceTrigger = {
+    device_id: "",
+    platform: "device",
+    domain: "",
+    entity_id: "",
+  };
+  private key?: string;
+  private setTrigger?: DeviceTrigger;
+
+  private readonly NO_TRIGGER = "NO_TRIGGER";
+  private readonly UNKNOWN_TRIGGER = "UNKNOWN_TRIGGER";
+
+  protected render(): TemplateResult | void {
+    const noTriggers = Object.keys(this.triggers).length === 0;
+    return html`
+      <paper-dropdown-menu-light .label=${this.label} ?disabled=${noTriggers}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._key}
+          attr-for-selected="key"
+          @iron-select=${this._triggerChanged}
+          id="listbox"
+        >
+          <paper-item .key=${this.NO_TRIGGER} .trigger=${this.noTrigger} hidden>
+            No triggers
+          </paper-item>
+          <paper-item
+            .key=${this.UNKNOWN_TRIGGER}
+            .trigger=${this.unknownTrigger}
+            hidden
+          >
+            Unknown trigger
+          </paper-item>
+          ${Object.keys(this.triggers).map(
+            (key) => html`
+              <paper-item .key=${key} .trigger=${this.triggers[key]}>
+                ${this.hass!.localize(
+                  `component.${
+                    this.triggers[key].domain
+                  }.device_automation.trigger_type.${this.triggers[key].type}`,
+                  "entity_id",
+                  this.triggers[key].entity_id
+                    ? computeStateName(
+                        this.hass!.states[this.triggers[key].entity_id]
+                      )
+                    : "",
+                  "event",
+                  this.triggers[key].event
+                )}
+              </paper-item>
+            `
+          )}
+        </paper-listbox>
+      </paper-dropdown-menu-light>
+    `;
+  }
+
+  private get _key() {
+    return this.key;
+  }
+
+  protected async firstUpdated() {
+    this.setTrigger = this.presetTrigger;
+  }
+
+  protected async updated(changedProps) {
```

Comment:
`updated` should not be async, as that causes things to be called out of order. Instead, call an async method from `updated`. `private async _updateDeviceInfo()`

Reply:
Hmm, maybe fixed?